packagehandlers/pnpmpackagehandler.go

```diff
@@ -0,0 +1,108 @@
+package packagehandlers
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/frogbot/v2/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"os"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	pnpmDependencyPattern    = "\\s*\"%s\"\\s*:\\s*\"[~|^]?%s\""
+	pnpmDescriptorFileSuffix = "package.json"
+	nodeModulesPathPattern   = ".*node_modules.*"
+)
+
+type PnpmPackageHandler struct {
+	CommonPackageHandler
+}
+
+func (pnpm *PnpmPackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
+	if vulnDetails.IsDirectDependency {
+		return pnpm.updateDirectDependency(vulnDetails)
+	}
+
+	return &utils.ErrUnsupportedFix{
+		PackageName:  vulnDetails.ImpactedDependencyName,
+		FixedVersion: vulnDetails.SuggestedFixedVersion,
+		ErrorType:    utils.IndirectDependencyFixNotSupported,
+	}
+}
+
+func (pnpm *PnpmPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails) (err error) {
+	descriptorFilesFullPaths, err := pnpm.CommonPackageHandler.GetAllDescriptorFilesFullPaths([]string{pnpmDescriptorFileSuffix}, nodeModulesPathPattern)
+	if err != nil {
+		return
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		err = fmt.Errorf("failed to get current working directory: %s", err.Error())
+		return
+	}
+
+	var anyDescriptorChanged bool
+	for _, descriptorFile := range descriptorFilesFullPaths {
+		var isFileChanged bool
+		isFileChanged, err = pnpm.fixVulnerabilityIfExists(vulnDetails, descriptorFile, wd)
+		if err != nil {
+			return
+		}
+		anyDescriptorChanged = anyDescriptorChanged || isFileChanged
+	}
+	if !anyDescriptorChanged {
+		err = fmt.Errorf("impacted package '%s' was not found in any descriptor files", vulnDetails.ImpactedDependencyName)
+	}
+	return
+}
+
+func (pnpm *PnpmPackageHandler) fixVulnerabilityIfExists(vulnDetails *utils.VulnerabilityDetails, descriptorFilePath, originalWd string) (isFileChanged bool, err error) {
+	var fileData []byte
+	fileData, err = os.ReadFile(descriptorFilePath)
+	if err != nil {
+		err = fmt.Errorf("failed to read file '%s': %s", descriptorFilePath, err.Error())
+		return
+	}
+
+	vulnDepRegexp := getRegexpCompilerForVulnerability(vulnDetails.ImpactedDependencyName, vulnDetails.ImpactedDependencyVersion)
+
+	// Only if the vulnerable dependency is detected in the current descriptor, we initiate a fix
+	if match := vulnDepRegexp.FindString(strings.ToLower(string(fileData))); match != "" {
+		modulePath := path.Dir(descriptorFilePath)
+		err = os.Chdir(modulePath)
+		if err != nil {
+			err = fmt.Errorf("failed to change directory to '%s': %s", modulePath, err.Error())
+			return
+		}
```

Comment:
Whenever possible, let's reduce nesting. 
```suggestion
		if err = os.Chdir(modulePath); err != nil {}
			return fmt.Errorf("failed to change directory to '%s': %s", modulePath, err.Error())
		}
```